README.md

```diff
@@ -1,2 +1,343 @@
+all of the underlying layers. This mode is used for Prints that are
+embossed into the material. An example would be to have an embossed
+logo.
+
+##### Replace All
+
+The *Replace All* blend mode is the simplest mode which overrides the
+normal map of the underlying Material.
+
+## Color Spaces
+
+In the AMDF the color space of constant colors need to be defined to
+correctly interpret the color values correctly. These color spaces have
+been restricted to "linear", "sRGB" and "AdobeRGB".
+
+## Texture Color Interpretation
+
+Textures can be used to store simple data (e.g. roughness maps), vectors
+(e.g. normal maps) or actual color data. Other than color textures, the
+values of simple data and vector maps should be used as they are and not
+be color corrected with the color space stored in the texture
+(none-color data). For this purpose, a *colorInterpretation* attribute
+exists for referenced textures, which can be either set to "ColorData"
+or "NoneColorData".
+
+## Parameter Functions
+
+### HSV Shift
+
+To recolor a virtual material layer, a colorization model has been
+proposed, which is capable of reproducing the effect that the same color
+on different materials comes out differently. The idea is as follows:
+-   A real material is colored using a given color (*originalColor* of a
+    Material).
+-   It is analyzed how the color looks on the material (*baseColor* of a
+    Material).
+-   The difference (e.g. less saturated and brighter) is cloned using a
+    given target color (*colorValue* of a Color).
+
+The pseudo code for this approach looks like this:
+
+```c++
+rgbColor recolorTexture(
+    Texture input,
+    rgbColor pdmColor,
+    rgbColor targetColor
+    TexCoord uv)
+{
+    unsigned int image_width = input.width;
+    unsigned int image_height = input.height; 
+
+    hsvColor hsv_pdm = RGBtoHSV(pdmColor);
+    hsvColor hsv_target = RGBtoHSV(targetColor);
+
+    // get the pixel color from the texture and
+    // convert it to HSV
+    rgbColor baseColor = tex2D(input, uv);
+    hsvColor hsv_base = RGBtoHSV(baseColor);
+
+    // compute hsv difference of pixel compared to pdmColor
+    hsvColor hsv_diff_to_pdm = hsv_base - hsv_pdm;
+
+    // kill difference in hue
+    hsv_diff_to_pdm.h = 0;
+
+    // apply difference on target color
+    hsvColor hsv_pix_target = hsv_target + hsv_diff_to_pdm;
+
+    // desaturate the output depending on the saturation 
+    // of the target color
+    hsvColor desat_hsv_pix_target = desatColor(hsv_pix_target, hsv_target);
+
+    rgbColor rgb_pix_target = HSVtoRGB(desat_hsv_pix_target);
+
+    return rgb_pix_target;
+}
+```
+
+As a slight red tinge was observed for desaturated target colors, the
+output color is desaturated using the following function:
+
+```c++
+hsvColor desatColor(
+    hsvColor recoloredPixel,
+    hsvColor targetColor)
+{
+    float s_t = targetColor.s;
+    float s_d = 1.0 - pow(1.0 - s_t, 5.0);
+
+    hsvColor c_d;
+	c_d.h = recoloredTexture.h;
```

Comment:
Is indentation correct?

Reply:
Normally blanks are used, but on those lines it added tabs somehow.